libc/resolve_path.c

```diff
@@ -475,6 +475,48 @@ TEST(resolve_path, symlink_rename)
 }
 
 
+/* check the behavior of the realpath function when it encounters pathname resolution of a symbolic link that result with a huge length */
+TEST(resolve_path, symlink_long_resolution)
+{
+	const char symNameShort[] = "symShort";
+	char symNameLong[NAME_MAX / 2];
+	char resolved[PATH_MAX + 1];
+	char basePath[PATH_MAX - 5] = "symShort/";
+	int basePathLen = strlen(basePath);
+	const char *pathSegment = "dev/../";
+	const int pathSegmentLen = sizeof(pathSegment);
```

Comment:
As you can see below using len name when assigning sizeof value may be misleading.

Reply:
yeah, for `sizeof` use name `pathSegmentSize`, in this case you could write `pathSegmentLen = sizeof(pathSegment) -1`